extensions/wikia/CuratedContent/CuratedContentSpecialController.class.php

```diff
@@ -124,58 +103,52 @@ public function index() {
 	}
 
 	public function featuredSection() {
-		$this->response->setTemplateEngine( self::TEMPLATE_ENGINE );
+		$this->response->setTemplateEngine( WikiaResponse::TEMPLATE_ENGINE_MUSTACHE );
 
-		$id = $this->request->getVal( 'image_id', 0 );
+		$imageId = $this->request->getVal( 'image_id', 0 );
 
 		$this->response->setVal( 'value', wfMessage( 'wikiacuratedcontent-featured-section-name' ) );
-		$this->response->setVal( 'image_id', $id );
-		$this->response->setVal( 'image_url', $this->getImage( $id ) );
-		if ( $id != 0 ) {
+		$this->response->setVal( 'image_id', $imageId );
+		$this->response->setVal( 'image_url', CuratedContentHelper::getImageUrl( $imageId ) );
+		if ( $imageId != 0 ) {
 			$this->response->setVal( 'image_set', true );
 		}
 	}
 
 	public function section() {
-		$this->response->setTemplateEngine( self::TEMPLATE_ENGINE );
+		$this->response->setTemplateEngine( WikiaResponse::TEMPLATE_ENGINE_MUSTACHE );
 
-		$id = $this->request->getVal( 'image_id', 0 );
+		$imageId = $this->request->getVal( 'image_id', 0 );
 
-		$this->response->setVal( 'value', $this->request->getVal( 'value' ), '' );
-		$this->response->setVal( 'image_id', $id );
-		$this->response->setVal( 'image_url', $this->getImage( $id ) );
-		if ( $id != 0 ) {
+		$this->response->setVal( 'value', $this->request->getVal( 'value' , '' ) );
+		$this->response->setVal( 'image_id', $imageId );
+		$this->response->setVal( 'image_url', CuratedContentHelper::getImageUrl( $imageId ) );
+		if ( empty( $imageId ) ) {
```

Comment:
you sometimes check if imageId != 0 (https://github.com/Wikia/app/pull/7862/files#diff-514f443fe83103e42561ed6c76b92b66R113) and sometimes if it is not empty. Can we unify it?